File: KSTests/Misc/JsonActionTests.vb

'    Kernel Simulator  Copyright (C) 2018-2022  EoflaOE
'
'    This file is part of Kernel Simulator
'
'    Kernel Simulator is free software: you can redistribute it and/or modify
'    it under the terms of the GNU General Public License as published by
'    the Free Software Foundation, either version 3 of the License, or
'    (at your option) any later version.
'
'    Kernel Simulator is distributed in the hope that it will be useful,
'    but WITHOUT ANY WARRANTY; without even the implied warranty of
'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'    GNU General Public License for more details.
'
'    You should have received a copy of the GNU General Public License
'    along with this program.  If not, see <https://www.gnu.org/licenses/>.

Imports System.IO
Imports KS.ConsoleBase
Imports KS.Misc.Beautifiers
Imports Newtonsoft.Json
Imports Newtonsoft.Json.Linq

<TestFixture> Public Class JsonActionTests

    ''' <summary>
    ''' Tests beautifying the JSON text
    ''' </summary>
    <Test, Description("Action")> Public Sub TestBeautifyJsonText()
        Dim Beautified As String = BeautifyJsonText(JsonConvert.SerializeObject(ColorDataJson))
        Beautified.ShouldNotBeEmpty
        Beautified.ShouldBe(JsonConvert.SerializeObject(ColorDataJson, Formatting.Indented))
    End Sub

    ''' <summary>
    ''' Tests beautifying the JSON text
    ''' </summary>
    <Test, Description("Action")> Public Sub TestBeautifyJsonFile()
        Dim SourcePath As String = Path.GetFullPath("Hacker.json")
        Dim Beautified As String = BeautifyJson(SourcePath)
        Beautified.ShouldNotBeEmpty
        Beautified.ShouldBe(JsonConvert.SerializeObject(JToken.Parse(Beautified), Formatting.Indented))
    End Sub

    ''' <summary>
    ''' Tests minifying the JSON text
    ''' </summary>
    <Test, Description("Action")> Public Sub TestMinifyJsonText()
        Dim Minified As String = MinifyJsonText(JsonConvert.SerializeObject(ColorDataJson))
        Minified.ShouldNotBeEmpty
        Minified.ShouldBe(JsonConvert.SerializeObject(ColorDataJson, Formatting.None))
    End Sub

    ''' <summary>
    ''' Tests minifying the JSON text
    ''' </summary>
    <Test, Description("Action")> Public Sub TestMinifyJsonFile()
        Dim SourcePath As String = Path.GetFullPath("Hacker.json")
        Dim Minified As String = MinifyJson(SourcePath)
        Minified.ShouldNotBeEmpty
        Minified.ShouldBe(JsonConvert.SerializeObject(JToken.Parse(Minified), Formatting.None))
    End Sub

End Class

File: KSTests/Misc/JsonActionTests.cs
//
// Kernel Simulator  Copyright (C) 2018-2024  Aptivi
//
// This file is part of Kernel Simulator
//
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY, without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// Kernel Simulator  Copyright (C) 2018-2022  Aptivi
// 
// This file is part of Kernel Simulator
// 
// Kernel Simulator is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// Kernel Simulator is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.

using System.IO;
using KS.ConsoleBase.Colors;
using KS.Misc.Beautifiers;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using NUnit.Framework;
using Shouldly;

namespace KSTests.Misc
{

    [TestFixture]
    public class JsonActionTests
    {

        /// <summary>
        /// Tests beautifying the JSON text
        /// </summary>
        [Test]
        [Description("Action")]
        public void TestBeautifyJsonText()
        {
            string Beautified = JsonBeautifier.BeautifyJsonText(JsonConvert.SerializeObject(Color255.ColorDataJson));
            Beautified.ShouldNotBeEmpty();
            Beautified.ShouldBe(JsonConvert.SerializeObject(Color255.ColorDataJson, Formatting.Indented));
        }

        /// <summary>
        /// Tests beautifying the JSON text
        /// </summary>
        [Test]
        [Description("Action")]
        public void TestBeautifyJsonFile()
        {
            string SourcePath = Path.GetFullPath("TestData/Hacker.json");
            string Beautified = JsonBeautifier.BeautifyJson(SourcePath);
            Beautified.ShouldNotBeEmpty();
            Beautified.ShouldBe(JsonConvert.SerializeObject(JToken.Parse(Beautified), Formatting.Indented));
        }

        /// <summary>
        /// Tests minifying the JSON text
        /// </summary>
        [Test]
        [Description("Action")]
        public void TestMinifyJsonText()
        {
            string Minified = JsonMinifier.MinifyJsonText(JsonConvert.SerializeObject(Color255.ColorDataJson));
            Minified.ShouldNotBeEmpty();
            Minified.ShouldBe(JsonConvert.SerializeObject(Color255.ColorDataJson, Formatting.None));
        }

        /// <summary>
        /// Tests minifying the JSON text
        /// </summary>
        [Test]
        [Description("Action")]
        public void TestMinifyJsonFile()
        {
            string SourcePath = Path.GetFullPath("TestData/Hacker.json");
            string Minified = JsonMinifier.MinifyJson(SourcePath);
            Minified.ShouldNotBeEmpty();
            Minified.ShouldBe(JsonConvert.SerializeObject(JToken.Parse(Minified), Formatting.None));
        }

    }
}